doc/sailfish-webview-qmozcontext.cpp

```diff
@@ -0,0 +1,367 @@
+    automatically set up on initialization, and can also be passed as
+    parameter to \l{WebEngine::initialize}.
+
+    When WebEngine is instantiated as a QML component this is also called
+    automatically during initalisation. However, it can be called earlier to
+    set a custom profile location in case the intialization process is being
+    overridden.
+
+    \note If using a different location care is needed to select a path that's
+    acceptable from an application sandboxing point of view. The qstandard
+    application paths are safe for profile paths.
+
+    See \l {SailfishOS::WebEngine::initialize}{WebEngine::initialize} for more
+    info.
+
+    \sa {SailfishOS::WebEngine::initialize}{WebEngine::initialize}, runEmbedding, stopEmbedding
+*/
+void QMozContext::setProfile(const QString &profilePath);
+
+/*!
+    \internal
+    \brief Called automatically on WebView creation so that the engine can create views.
+
+    For internal use.
+
+    \sa setViewCreator
+*/
+void QMozContext::setViewCreator(QMozViewCreator *viewCreator);
+
+/*!
+    \internal
+    \brief Opens a new view that can contain webcontent.
+
+    For internal use.
+
+    \sa setViewCreator
+*/
+void QMozContext::createView();
+
+/*!
+    \brief This signal is emitted when a notification on an observed topic is received.
+
+    After an app has registered an interest in a particular topic by calling
+    \l addObserver or \l addObservers, it can then conect to this signal to be
+    notified about any component sending a notification on the topic.
+
+    The \a message parameter will be the topic in question, the \a data parameter
+    will contain data sent by the sender constructed from a JSON object. The
+    exact object structure is at the sender's discretion.
+
+    \sa addObserver, addObservers, removeObserver, removeObservers, notifyObservers
+*/
+void QMozContext::recvObserve(const QString message, const QVariant data);
+
+/*!
+    \brief This signal is emitted when the gecko EmbedLiteApp has been destroyed.
+
+    This signal is emitted when the EmbedLiteApp has been destroyed, just prior
+    to the WebEngine being deleted and the program exiting.
+*/
+void QMozContext::contextDestroyed();
+
+/*!
+    \brief This signal is emitted when when the last view has been destroyed.
+*/
+void QMozContext::lastViewDestroyed();
+
+/*!
+    \brief This signal is sent when the last registered window has been destroyed.
+
+    By default this is used to delay execution of \l stopEmbedding until after
+    the last window has closed.
+*/
+void QMozContext::lastWindowDestroyed();
+
+/*!
+    \brief Regsiter javascript chrome components to be loaded into the WebEngine.
+
+    The sailfish WebEngine has three main layers. The top layer is exposed as
+    QML components with a developer-facing interface. The lowest level is the
+    native gecko rendering engine.
+
+    Between the two sits a javascript chrome layer that can communicate with
```

Comment:
What is meant by "chrome layer"?  I'd prefer something more expressive, maybe "a content and event processing layer defined in JavaScript" but I'm not sure.

Reply:
I guess I was thinking this is where all the XUL stuff used to happen. Rather than wrestle with what it really is, I've just changed it to "JavaScript layer".